packages/storage-plugin/schematics/src/ng-generate/index.ts

```diff
@@ -0,0 +1,91 @@
+import { exit } from 'node:process';
+import * as ts from 'ts-morph';
+
+import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
+import { visitTsFiles } from '../utils/visit-files';
+
+function migrateEmptyForRoot(callExpression: ts.CallExpression<ts.ts.CallExpression>) {
+  callExpression.addArgument(`{ keys: '*' }`);
+}
+
+function migrateForRootWithArgs(arg: ts.Node<ts.ts.Node>) {
+  const objectLiteral = arg as ts.ObjectLiteralExpression;
+
+  const hasPropertyName = (name: string) =>
+    objectLiteral.getProperties().some(prop => {
+      return ts.Node.isPropertyAssignment(prop) && prop.getName() === name;
+    });
+
+  const isAlreadyMigrated = hasPropertyName('keys');
+
+  // If the "keys" property is already there, we should not try migrating
+  if (isAlreadyMigrated) {
+    return;
+  }
+
+  const hasKeyProperty = hasPropertyName('key');
+
+  // Add the "keys" property if it's not there
+  if (!hasKeyProperty) {
+    objectLiteral.addPropertyAssignment({ name: 'keys', initializer: `'*'` });
+  } else {
+    // Rename the "key" property to "keys"
+    objectLiteral.getProperty('key')?.getChildAtIndex(0)?.replaceWithText('keys');
+  }
+}
+
+function isStoragePluginProvided(callExpression: ts.CallExpression) {
+  return (
+    callExpression && callExpression.getText().includes('NgxsStoragePluginModule.forRoot')
+  );
+}
+
+export function migrateKeys(): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const tsProject = new ts.Project({ useInMemoryFileSystem: true });
+    const packageJson = tree.readJson('package.json') as any;
+    const storePackage = packageJson['dependencies']['@ngxs/store'];
+
+    if (!storePackage) {
+      _context.logger.error(`No @ngxs/store found`);
+      return exit(1);
+    }
+
+    visitTsFiles(tree, tree.root, async path => {
+      const fileContent = tree.readText(path);
+      const sourceFile = tsProject.createSourceFile(path, fileContent!);
+      const hasStoragePluginImported = sourceFile.getImportDeclaration(
+        importDecl => importDecl.getModuleSpecifierValue() === '@ngxs/storage-plugin'
+      );
+
+      // do not try migrating if the storage plugin is not imported
+      if (hasStoragePluginImported) {
+        sourceFile.forEachDescendant(node => {
+          if (
+            ts.Node.isCallExpression(node) &&
```

Comment:
I'd rather do something like
`node.getKindName() === 'CallExpression'`

That will be safer because under the hood this guy does something like `node.getKind() === typescript.SyntaxKind.CallExpression`, which is an enum that can return different numbers depending on minor typescript package versions. And ts-morph kindly does not set typescript as a dependency

Reply:
Thanks for explaining this. I did the change